test/cast/object_test.exs

```diff
@@ -337,6 +347,19 @@ defmodule OpenApiSpex.ObjectTest do
       assert user == %User{name: "Name"}
     end
 
+    test "casts to struct even if input was a different struct" do
+      schema = %Schema{
+        type: :object,
+        "x-struct": User,
+        properties: %{
+          name: %Schema{type: :string}
+        }
+      }
+
+      assert {:ok, user} = cast(value: %User2{name: "Name"}, schema: schema)
+      assert user == %User{name: "Name"}
```

Comment:
Let's add another test case for when `x-struct` is not set and the input is a `User` struct, it should be preserved.

Reply:
Okay, test added (should be what you meant)